libc-bottom-half/sources/posix.c

```diff
@@ -351,6 +351,13 @@ int fstatvfs(int fd, struct statvfs *buf) {
     return -1;
 }
 
+char *realpath (const char *__restrict file_name, char *__restrict resolved_name) {
+    // TODO: We plan to support this eventually in WASI, but not yet.
+    // Meanwhile, we provide a stub so that libc++'s `<filesystem>`
+    // implementation will build unmodified.
+    return NULL;
```

Comment:
realpath is supposed to set `errno` when it fails.  

I'm not sure which one would make sense here.  The man pages says:

```
ERRORS
       EACCES Read or search permission was denied for a component of the path prefix.

       EINVAL path is NULL.  (Before glibc 2.3, this error is also returned  if  resolved_path  is
              NULL.)

       EIO    An I/O error occurred while reading from the filesystem.

       ELOOP  Too many symbolic links were encountered in translating the pathname.

       ENAMETOOLONG
              A  component  of  a pathname exceeded NAME_MAX characters, or an entire pathname ex‐
              ceeded PATH_MAX characters.

       ENOENT The named file does not exist.

       ENOMEM Out of memory.

       ENOTDIR
              A component of the path prefix is not a directory.

```

Reply:
Good catch -- sorry I missed that.  I was looking at the `RETURN VALUES` section of the macOS man page, which didn't mention `errno`, but failed to read the `ERRORS` section.  In any case, I've updated this PR to use musl's implementation and removed the stub.